components/cards/SessionCard.tsx

```diff
@@ -0,0 +1,275 @@
+ * @param handleBookMark: function that bookmarks and returns nothing
+ * @param item: object of type T
+ * @param screen: 'home' | 'sessions', default is 'home'
+ * @param variant: 'card' | 'list'
+ * @returns SessionCard Component
+ */
+
+const SessionCardOnHome = (props: Omit<SessionCardHome, 'screen'>) => {
+  const { handlePress, item } = props;
+  const { colors } = useTheme();
+
+  return (
+    <TouchableWithoutFeedback testID="card" onPress={handlePress} disabled={item.slug.length < 1}>
+      <View style={[styles.card, { backgroundColor: colors.card }]}>
+        <Image
+          source={{ uri: item.session_image || '' }}
+          style={styles.image}
+          contentFit="cover"
+          contentPosition="left"
+        />
+        <Space size={8} />
+        <View style={styles.bottom}>
+          <View style={styles.description}>
+            <StyledText font="bold" numberOfLines={2} style={styles.title}>
+              {truncate(50, item.title)}
+            </StyledText>
+          </View>
+
+          <Space size={12} />
+
+          <StyledText size="sm" font="light">
+            {getSessionTimeAndLocation(item.slug)}
+          </StyledText>
+        </View>
+      </View>
+    </TouchableWithoutFeedback>
+  );
+};
+
+const SessionCardOnSessions = (props: Omit<SessionCardSessions, 'screen'>) => {
+  const { handlePress, item, handleBookMark } = props;
+  const { colors } = useTheme();
+  const router = useRouter();
+  return (
+    <TouchableWithoutFeedback testID="card-sessions" onPress={handlePress} disabled={item.slug.length < 1}>
+      <View style={[styles.card, styles.cardTertiary, { backgroundColor: colors.card }]}>
+        {item.session_image !== null ? (
+          <Image
+            source={{
+              uri:
+                item.session_image ||
+                'https://res.cloudinary.com/droidconke/image/upload/v1668083311/prod/upload/sessions/pwt9pnojtmng8bymxkls.png',
+            }}
+            style={styles.image}
+            contentFit="cover"
+            contentPosition="left"
+          />
+        ) : (
+          <Image
+            source={require('../../assets/images/banner.png')}
+            style={styles.image}
+            contentFit="cover"
+            contentPosition="center"
+          />
+        )}
+
+        <Space size={4} />
+
+        <View style={styles.bottomForSessions}>
+          <StyledText size="sm" font="light">
+            {getScheduleTimeAndLocation(item.start_date_time, item.end_date_time, item.rooms[0])}
+          </StyledText>
+          <Space size={11} />
+          <View>
+            <StyledText font="bold" numberOfLines={2} style={styles.title}>
+              {truncate(50, item.title)}
+            </StyledText>
+          </View>
+          <Space size={16} />
+          <Row>
+            <Row style={styles.avatarRow}>
+              {item.speakers.map((speaker) => (
+                <Pressable
+                  key={speaker.avatar}
```

Comment:
since `[speaker]` is a dynamic route, you don't need to navigate to `/speaker`. Ideally the best way would be to use the `Link` component. but with `router` you can still navigate to speaker as below

```
  onPress={() => router.push(`/{speaker.id}`)}
```
now with the above, each speaker id generates a dynamic route, and now in the speaker page we can fetch speaker details.

you could alternatively wrap the Pressable with a `Link` component as below:

```
import { Link } from 'expo-router';

<Link href={`/{speaker.id}`} key={speaker.avatar}>
  <Pressable>
    <Image source={{ uri: speaker.avatar || '' }} style={[styles.avatar, { borderColor: colors.tint }]} />
  </Pressable>
</Link>
```

remember if you want to replace the view, the `useRouter` is the best way to go, but if its just pushing a stack view, then `Link` works

Reply:
Alright, I now see the difference between Link and router. 
I passed the whole speaker object because I noticed there was no endpoint for fetching a single speaker. How best can we handle this scenario? Btw, I have just invited you to the Postman APIs Collection for the app via the email antolisah@gmail.com

Reply:
I think you missed the point I was trying to make. Right now we don't have a `/speaker` route. Ideally its a dynamic route which is unmatched. so to access a speaker route we call `/${speaker.<a unique identifier>}`. I think for now you can either pass the `name` parameter or leave it as an empty function until we figure out the best way to do this. my advice try passing the `name` as below:

```
<Link key={speaker.avatar} href={{ pathname: `/${speaker.name}`, params: speaker }}>
```

and test if it works

Reply:
So you literally parse anything to that screen? ok, I think I now get it. I will use speaker's name as identifier for now.